aptos-move/block-executor/src/executor.rs

```diff
@@ -1084,6 +1101,9 @@ where
         env: &E::Environment,
         signature_verified_block: &[T],
         base_view: &S,
+        global_module_cache: &RwLock<
```

Comment:
if this isn't already a rw lock guard (already locked), when we lock, it may no longer be the correct thing.

Reply:
It cannot be a guard? We need to keep at possible mutable, because in the end of the block we sync modules from mvhashmap into it?

Reply:
we could return published modules from block execution and apply to the cache in the wrapper after upgrading the lock to write (fancier, or re-locking). This way the same RWLock can be the whole synchronization required for manager. However, if we want smaller change, we could just fix the invariant in the wrapper for now - but guard or not it seems the manager can/should be simplified. 
Another way would be to provide block hash and do the checks here while locking instead of the wrapper.